### PR TITLE
Extending detection object

### PR DESCRIPTION
**Pull request by matthastings**

### CODE REVIEW

**Comment by matthastings** [comment]
@aztechian Can you give this a quick 👀

File: python/redcanary/detectors.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+class Detector(object):
+    """
+    Object class for a red canary detector
```
Comment (aztechian) [inline python/redcanary/detectors.py:3]:
```suggestion
    Object class for a Red Canary detector
```

File: python/redcanary/detections.py [hunk 2]
```diff
@@ -57,15 +72,64 @@ def all(self, per_page: int = None, since: str = None) -> list:
         """
         return [Detection(i) for i in self._get_all('detections', locals())]
 
-    def byid(self, id: int) -> object:
+    def get_by_id(self, id: int) -> object:
```
Comment (aztechian) [inline python/redcanary/detections.py:75]:
```suggestion
    def by_id(self, id: int) -> object:
```
I'd suggest somewhere in the middle ;)
The `get_` seems superfluous. Just my opinion...

File: python/redcanary/detections.py [hunk 3]
```diff
@@ -57,15 +72,64 @@ def all(self, per_page: int = None, since: str = None) -> list:
...
+        Searches for detection by id
+
         Parameters
         --------
         id : int
             detection id
         """
         return Detection(self._get_by_id('detections', id))
 
+    def get_detectors(self, detection: object) -> list:
```
Comment (aztechian) [inline python/redcanary/detections.py:86]:
```suggestion
    def detectors(self, detection: object) -> list:
```
Same as above comment

File: python/redcanary/detections.py [hunk 4]
```diff
@@ -40,12 +41,26 @@ def __init__(self, customer_id: str = None, auth_token: str = None):
...
+        self._str_remediation_options = [
+            "remediated",
+            "not_remediated_false_positive",
+            "not_remediated_sanctioned_activity",
+            "not_remediated_unwarranted"
+        ]
+        self.REMEDIATED = self._str_remediation_options[0]
+        self.FALSE_POSITIVE = self._str_remediation_options[1]
+        self.SANCTIONED_ACTIVITY = self._str_remediation_options[2]
+        self.UNWANTED = self._str_remediation_options[3]
```
Comment (aztechian) [inline python/redcanary/detections.py:54]: Not that its wrong - this just reads really weird.
I'd set the "static" string variables with the literal, then make the list from those.
Also, less chance of mixing array indexes when this list changes in the future.

**Review by aztechian — APPROVED** [review]

File: python/redcanary/detections.py (hunk 2 above)
Comment (matthastings) [reply python/redcanary/detections.py:75]: I'm used to PowerShell where everything starts with a verb

File: python/redcanary/detections.py (hunk 4 above)
Comment (matthastings) [reply python/redcanary/detections.py]: Nod, good point. Updated based on your recommendation

**Pull request merged**